elixir/apps/domain/priv/repo/migrations/20250227193411_convert_dns_ip_resources_to_ip.exs

```diff
@@ -0,0 +1,37 @@
+defmodule Domain.Repo.Migrations.ConvertDnsIpResourcesToIp do
+  use Ecto.Migration
+
+  def change do
+    # Convert IPv4 addresses
+    execute("""
+    UPDATE resources
+    SET type = 'ip'
+    WHERE type = 'dns'
+      AND address ~ '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
+    """)
+
+    # Convert IPv6 addresses
+    execute("""
+    UPDATE resources
+    SET type = 'ip'
+    WHERE type = 'dns'
+      AND address ~ '^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$'
+    """)
```

Comment:
That's quite the regex 😆 

Would it be as effective to use something like this:
```
UPDATE resources
    SET type = 'ip'
    WHERE type = 'dns'
    AND address ~ '^[0-9a-fA-F:]+$' 
    AND address LIKE '%:%';
```

Reply:
Ah, I just realized you have another similar one looking for IPv6 CIDR blocks below.  

Here's an updated suggestion:
```
UPDATE resources
    SET type = 'ip'
    WHERE type = 'dns'
    AND address ~ '^[0-9a-fA-F:]+$' 
    AND address LIKE '%:%'
    AND address NOT LIKE '%/%';
```

Reply:
Yeah, this is some crap "AI" spit out. Garbage.

We can simply try to cast to inet.

```
      UPDATE resources
      SET type = 'ip'
      WHERE type = 'dns'
        AND address::inet IS NOT NULL
        AND address !~ '/'
```